lib/plugins/matrix.py

```diff
@@ -49,19 +49,28 @@ async def async_format_content(self, content, mentions, hashtags, images, **kwar
                 # try to get the display name of the mentioned matrix user
                 response = await self.client.get_displayname(f"@{mention}")
                 mention_name = getattr(response, "displayname", mention)
-                mention_links.append(f"[{mention_name}](https://matrix.to/#/@{mention})")
+                mention_links.append(
+                    f"[{mention_name}](https://matrix.to/#/@{mention})"
+                )
                 message_content["m.mentions"]["user_ids"].append(f"@{mention}")
             mentions_string = " ".join(mention_links)
             content = f"{mentions_string}: {content}"
         if hashtags:
             content += "\n\n" + " ".join([f"\\#{h}" for h in hashtags])
         formatted_body = markdown(content)
-        body = BeautifulSoup(formatted_body, features="html.parser").get_text("\n", strip=True)
+        body = BeautifulSoup(formatted_body, features="html.parser").get_text(
+            "\n", strip=True
+        )
         message_content["body"] = body
         message_content["formatted_body"] = formatted_body
         formatted_content.append(message_content)
         warnings = ""
-        return formatted_content, preview + "\n" + message_content["formatted_body"], warnings
+        await self.client.close()
```

Comment:
@arash77 do you know why this works?

I know why you added the line, but wouldn't it mean that the client is closed when create_post runs?

Reply:
Looks like most Client methods ensure there's an open session before doing anything so create_post simply opens a session again :-)

This might not be the most efficient way to do things then, but we can address that later.